nats.go

```diff
@@ -3623,9 +3625,45 @@ func (nc *Conn) unsubscribe(sub *Subscription, max int, drainMode bool) error {
 	return nil
 }
 
-func (s *Subscription) handleControlMessage(msg *Msg) (bool, error) {
+// ErrConsumerSequenceMismatch represents an error where the consumer sequence.
+type ErrConsumerSequenceMismatch struct {
+	// StreamResumeSequence is the stream sequence from where the consumer
+	// should resume consuming from the stream.
+	StreamResumeSequence uint64
```

Comment:
Should we put in the consumer sequence too in case they want access to it for some reason?